### DISCUSSION TITLE
Using augmented-ui for CSS Themes in Obsidian.md

### DISCUSSION THREAD

**Discussion by TonkatsuRa** [propjockey/augmented-ui#16] [Q&A] [answered]
I wanted to know if there is any way to use augmented-ui for making a cyberpunk theme for my obsidian vault.
The problem I see is, that the obsidian themes only use CSS code or CSS snippets to adjust the looks. So far I always had to add "data-augmented-ui" in the HTML code for it to work.

An example would be the ITS-Theme: https://github.com/SlRvb/Obsidian--ITS-Theme 

Any solution to this?

**Comment by JaneOri** [comment]
I see... There's not currently (there's work in progress that's a long way off still that would help... but not currently useful)

Unless... Are you hosting it on a custom domain? In that case, it also allows a `publish.js`
That js file can add `data-augmented-ui` attribute to any elements programmatically.

https://help.obsidian.md/Obsidian+Publish/Customize+your+site

Thoughts?

**Reply by TonkatsuRa** [reply]
No, I don't use the premium publish feature. I just got my vault on my google drive and sync it that way between devices.
I guess I would first need to create a plugin for obsidian that hosts JS code for the themes locally, before augmented-ui would be possible?

Unless it would somehow work with JSON code. Then I might be able to figure out a way.

Edit: From what I saw, the plugins for Obsidian are in JS code.

**Reply by JaneOri** [reply]
Hmm... I could whip up a more global augmented-ui file for you. Is there a set of elements/class names that you want `augmented-ui` to operate on for Obsidian usage?

Like every `div.container > p` should be augmented, etc - could you make that list?

You would have to use the core properties directly / wouldn't be able to use the mixins... full range of augmented-ui is possible with core alone.

**Reply by TonkatsuRa** [reply]
Hi JaneOri
Thank you so much for your offer. I will try to make a list but my knowledge is rather limited. Give me some time to figure everything out and I will come back to you.

I've never made a plugin or custom theme (only single snippets) so far and would need to dig into different Themes and see what they do / use to figure out what I need.

I forwarded this thread to the creator of ITS-Theme  @SlRvb in hope he will join me and help me figure this out.

**Reply by TonkatsuRa** [reply, marked as answer]
Hi @JaneOri 
I did some research and asked other devs of plugins for Obsidian.

I would need to write my own plugin to make it work but I am not skilled enough for that.

So I need to put that project aside for the moment until I understand JS code better

![Screenshot_2023-05-06-19-05-01-662_com.github.android.jpg](https://user-images.githubusercontent.com/115587388/236637601-9407e544-0aa1-4c3c-8def-272c47f51361.jpg)

**Comment by JaneOri** [comment]
No worries! Tyvm for the interest and introducing me to Obsidian. Please do reopen (or open another issue) if you ever dive in :)

Stay well!

**Comment by JaneOri** [comment]
I have recently begun using Obsidian and it's pretty awesome. I'm re-opening this as a reminder to myself for future investigation and consideration - though it may be quite some time.

If you had designs in mind, I'd be happy and interested to take specific notes too
